src/plugins/explore/public/application/utils/languages/ppl/get_query_string_with_source/get_query_with_source.ts

```diff
@@ -16,12 +16,26 @@ export const getQueryWithSource = (query: Query): QueryWithQueryAsString => {
   const hasDescribe = /^\s*describe\s+/.test(lowerCaseQuery);
   const hasShow = /^\s*show\s+/.test(lowerCaseQuery);
 
+  // Temporarily adding backticks to dataset type INDEXES or INDEX_PATTERNS to until these two issues are resolved:
+  // https://github.com/opensearch-project/sql/issues/4444
+  // https://github.com/opensearch-project/sql/issues/4445
+  let datasetTitle: string;
+  if (query.dataset && ['INDEXES', 'INDEX_PATTERN'].includes(query.dataset.type)) {
+    if (hasSource) {
+      // Replace source=anything with source=`anything` (handling spaces around = and ensuring anything is not already backticked)
+      const updatedQuery = queryString.replace(/(\bsource\s*=\s*)([^`\s][^\s|]*)/gi, '$1`$2`');
+      return { ...query, query: updatedQuery };
+    }
```

Comment:
do we even need this? assuming most users will directly select dataset rather than custom query, they won't fall into this if. If user lands here, they manually write the PPL query and I feel we shouldn't touch it?